### ISSUE TITLE
Portal dialog frozen and no mouse cursor after Hilda update

### ISSUE THREAD

**Issue by tdrury** [SpikeHimself/XPortal#61]
**Describe the bug**
When  you press 'e' to bring up the portal dialog, you cannot interact with it with the mouse. You can use the keyboard to change Name and Destination, but there's no way to close the dialog. 

![image](https://github.com/SpikeHimself/XPortal/assets/294530/d8056c77-8de7-4fc7-a163-8475edcf12c9)

**To Reproduce**
See above

**Expected behaviour**
Describe what you expected to happen (if it's not obvious).
Narrator voice: it's obvious.

**LogOutput file(s)**

```
[Debug  :XPortal.Log] [XPortal.OnPortalRequestText]  Interacting with: { Id: `1:481754`, Name; `_base`, Location: `(-5686.5, 35.8, 1672.7)`, Target: `(None)`, Colour: `#FF6400` }
TypeLoadException: Could not resolve type with token 01000091 (from typeref, class/assembly UIGamePad, assembly_guiutils, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null)
  at XPortal.UI.PortalConfigurationPanel.InitialiseUI () [0x0041d] in <e556ccad14a3493fbc4b90983a56316e>:0 
  at XPortal.UI.PortalConfigurationPanel.ConfigurePortal (XPortal.KnownPortal portal) [0x00000] in <e556ccad14a3493fbc4b90983a56316e>:0 
  at XPortal.XPortal.OnPortalRequestText (ZDOID portalId) [0x00039] in <e556ccad14a3493fbc4b90983a56316e>:0 
  at XPortal.Patches.TextInput_RequestText.Prefix (TextReceiver sign) [0x00015] in <e556ccad14a3493fbc4b90983a56316e>:0 
  at (wrapper dynamic-method) TextInput.DMD<TextInput::RequestText>(TextInput,TextReceiver,string,int)
  at TeleportWorld.Interact (Humanoid human, System.Boolean hold, System.Boolean alt) [0x00033] in <447b19510f604ca3807cd6cce3a7c1ef>:0 
  at (wrapper dynamic-method) Player.DMD<Player::Interact>(Player,UnityEngine.GameObject,bool,bool)
  at (wrapper dynamic-method) Player.DMD<Player::Update>(Player)

08/24/2023 13:34:36: Setting selected recipe 0
ExtendedPlayerInventory InventoryGui Show
Fallback handler could not load library D:/SteamLibrary/steamapps/common/Valheim/valheim_Data/Mono/data-000001AC062F7C20.dll
08/24/2023 13:34:40: Setting selected recipe 0
ExtendedPlayerInventory InventoryGui Show
08/24/2023 13:34:42: Setting selected recipe 0
ExtendedPlayerInventory InventoryGui Show
[Debug  :XPortal.Log] [XPortal.OnPortalRequestText]  Interacting with: { Id: `1:481754`, Name; `_base`, Location: `(-5686.5, 35.8, 1672.7)`, Target: `(None)`, Colour: `#FF6400` }
NullReferenceException: Object reference not set to an instance of an object
  at XPortal.UI.PortalConfigurationPanel.SetPingMapButtonActive (System.Boolean active) [0x00025] in <e556ccad14a3493fbc4b90983a56316e>:0 
  at XPortal.UI.PortalConfigurationPanel.PopulateDropdown () [0x001f9] in <e556ccad14a3493fbc4b90983a56316e>:0 
  at XPortal.UI.PortalConfigurationPanel.ConfigurePortal (XPortal.KnownPortal portal) [0x0002a] in <e556ccad14a3493fbc4b90983a56316e>:0 
  at XPortal.XPortal.OnPortalRequestText (ZDOID portalId) [0x00039] in <e556ccad14a3493fbc4b90983a56316e>:0 
  at XPortal.Patches.TextInput_RequestText.Prefix (TextReceiver sign) [0x00015] in <e556ccad14a3493fbc4b90983a56316e>:0 
  at (wrapper dynamic-method) TextInput.DMD<TextInput::RequestText>(TextInput,TextReceiver,string,int)
  at TeleportWorld.Interact (Humanoid human, System.Boolean hold, System.Boolean alt) [0x00033] in <447b19510f604ca3807cd6cce3a7c1ef>:0 
  at (wrapper dynamic-method) Player.DMD<Player::Interact>(Player,UnityEngine.GameObject,bool,bool)
  at (wrapper dynamic-method) Player.DMD<Player::Update>(Player)
```

**Version**
1.2.13

**Environment**

* Singleplayer

**Other mods**

[Info   :   BepInEx] 52 plugins to load
[Info   :   BepInEx] Loading [PlantEasily 1.6.1]
Fallback handler could not load library D:/SteamLibrary/steamapps/common/Valheim/valheim_Data/Mono/data-000001A9B611A2F0.dll
[Info   :   BepInEx] Loading [PlantEverything 1.13.7]
[Info   :   BepInEx] Loading [Build Restriction Tweaks 0.3.0]
[Info   :   BepInEx] Loading [Extended Player Inventory 0.6.3]
[Info   :   BepInEx] Loading [Hoe Radius 0.1.2]
[Info   :   BepInEx] Loading [Instant Monster Drop 0.4.0]
[Info   :   BepInEx] Loading [Terrain Reset 0.7.0]
[Info   :   BepInEx] Loading [Valheim - Silence Tame Wolf Cub Howls 0.0.6]
[Info   :   BepInEx] Loading [AzuAreaRepair 1.1.1]
[Info   :   BepInEx] Loading [AzuClock 1.0.0]
[Info   :   BepInEx] Loading [AzuCraftyBoxes 1.2.0]
[Info   :   BepInEx] Loading [BuildCameraCHE 1.0.0]
[Info   :   BepInEx] Loading [FirstPersonMode 1.2.2]
[Info   :   BepInEx] Loading [NoBuildDust 1.0.3]
[Info   :   BepInEx] Loading [NoCultivatorDust 1.0.3]
[Info   :   BepInEx] Loading [NoHoeDust 1.0.3]
[Info   :   BepInEx] Loading [Recycle_N_Reclaim 1.1.5]
[Info   :   BepInEx] Loading [FoodDurationMultiplier 1.0.8]
[Info   :   BepInEx] Loading [AutoPicker 1.1.3]
[Info   :   BepInEx] Loading [CaveSoundsBeGone 1.0.0]
[Info   :   BepInEx] Loading [Configuration Manager 18.0.1]
[Info   :   BepInEx] Loading [Jotunn 2.12.5]
[Info   :   BepInEx] Loading [MissingPieces 2.0.2]
[Info   :   BepInEx] Loading [TeleportEverything 2.5.0]
[Info   :   BepInEx] Loading [Use Equipment in Water 0.2.4]
[Info   :   BepInEx] Loading [Pressure-Plate 0.9.2]
[Info   :   BepInEx] Loading [VentureFloatingItems 0.1.1]
[Info   :   BepInEx] Loading [Skills Give More Carry Weight 1.0.3]
[Info   :   BepInEx] Loading [Veinmine 1.2.8]
[Info   :   BepInEx] Loading [Better Wisps 1.0.17]
[Info   :   BepInEx] Loading [Craftable Chain 1.0.0.0]
[Info   :   BepInEx] Loading [QuickTeleport 1.0.0]
[Info   :   BepInEx] Loading [DiscoveryRadius 1.0.1]
[Info   :   BepInEx] Loading [More Gates 1.0.10]
[Info   :   BepInEx] Loading [Item Drawers 0.5.6]
[Info   :   BepInEx] Loading [PlanBuild 0.14.2]
[Info   :   BepInEx] Loading [FuelEternal 1.2.1]
[Info   :   BepInEx] Loading [QuietPortals 0.1.4]
[Info   :   BepInEx] Loading [Gravekeeper 2.2.0]
[Info   :   BepInEx] Loading [Creature Level & Loot Control 4.5.11]
[Info   :   BepInEx] Loading [Sailing 1.1.6]
[Info   :   BepInEx] Loading [Display BepInEx Info In-Game 2.0.1]
[Info   :   BepInEx] Loading [OdinArchitect 1.2.7]
[Info   :   BepInEx] Loading [BetterLanterns 1.0.5]
[Info   :   BepInEx] Loading [SearsCatalog 1.2.0]
[Info   :   BepInEx] Loading [Render Limits 1.8]
[Info   :   BepInEx] Loading [AutomaticFuel 1.3.8]
[Info   :   BepInEx] Loading [Shield Auto-Equip 1.0.0]
[Info   :   BepInEx] Loading [AutoRepair 5.4.1600]
[Info   :   BepInEx] Loading [DvergrPieces 2.1.0]
[Info   :   BepInEx] Loading [No Fog Bruh 1.0.4]
[Info   :   BepInEx] Loading [XPortal 1.2.13]

**Priority**
Estimate how important this problem is. Choose Medium if you don't know.

* High (XPortal does not work at all anymore)

**Comment by CharlieTheQ** [comment]
I reproduced this problem on a world with no other mods besides XPortal and its required parents Jotunn and BepInExPack_Valheim, in case that helps at all, as well as my main one with many mods.  I was in the process of putting together a bug report when I saw this one posted.  It's completely broken and right now the only thing keeping me from the Hildir update (all of my portals become inaccessible if I disable this mod because their names are too long for vanilla apparently).  Hopefully you're still active!

**Comment by barballs** [comment]
Same issue. Windows host. Bare metal.  No other mods outside of dependents.

**Comment by lessagg** [comment]
I have the same problem here, mate.. can you please do something?

**Comment by LyndonL37** [comment]
Confirmed same issue with latest Jotunn 2-12-6 installed. Moving from 2-12-3 to -6 fixed the pink background, but the buttons are still missing and no UI focus.

No other mods except ClockMod (plus Jotunn/BepInEx).

```
[Debug  :XPortal.Log] [XPortal.OnPortalRequestText]  Interacting with: { Id: `1:314042`, Name; `Main Base`, Location: `(-109.7, 32.9, 57.4)`, Target: `(None)`, Colour: `#FF6400` }
[Error  : Unity Log] TypeLoadException: Could not resolve type with token 01000091 (from typeref, class/assembly UIGamePad, assembly_guiutils, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null)
Stack trace:
XPortal.UI.PortalConfigurationPanel.InitialiseUI () (at <e556ccad14a3493fbc4b90983a56316e>:0)
XPortal.UI.PortalConfigurationPanel.ConfigurePortal (XPortal.KnownPortal portal) (at <e556ccad14a3493fbc4b90983a56316e>:0)
XPortal.XPortal.OnPortalRequestText (ZDOID portalId) (at <e556ccad14a3493fbc4b90983a56316e>:0)
XPortal.Patches.TextInput_RequestText.Prefix (TextReceiver sign) (at <e556ccad14a3493fbc4b90983a56316e>:0)
(wrapper dynamic-method) TextInput.DMD<TextInput::RequestText>(TextInput,TextReceiver,string,int)
TeleportWorld.Interact (Humanoid human, System.Boolean hold, System.Boolean alt) (at <447b19510f604ca3807cd6cce3a7c1ef>:0)
Player.Interact (UnityEngine.GameObject go, System.Boolean hold, System.Boolean alt) (at <447b19510f604ca3807cd6cce3a7c1ef>:0)
Player.Update () (at <447b19510f604ca3807cd6cce3a7c1ef>:0)
```

**Comment by Sin-Thor** [comment]
Same problem.

I can't close the box so I have to completely close the game.

**Comment by Quentin-ctrL** [comment]
Jotunn release recently v2.12.6 
Did it solve the issue ?

**Comment by tdrury** [comment]
> Jotunn release recently v2.12.6 Did it solve the issue ?

No.

**Comment by amazingornaldo** [comment]
I too am having this issue. I did update Jotunn and it is no longer the nice pink anymore. So progress, but still cannot interact with the window and have to kill my Valheim process to exit the game and close the window.

**Comment by SpikeHimself** [comment]
Sorry for the delay everyone, life has been busy over here. I have just released v1.2.14 which should address this issue. If it has not, or you find other issues, please let me know. I will leave this issue open for now.

**Comment by tdrury** [comment]
It works! No errors in the logs. Thank you - this is the best portal mod in all of Valheim.

**Comment by SpikeHimself** [comment]
Thanks for reporting that @LyndonL37. Would you mind creating a new issue for that? I'll try look into it soon.

**Comment by SpikeHimself** [comment]
Confirmed fixed by v1.2.14
